bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/SchemaRepository.java

```diff
@@ -0,0 +1,49 @@
+/********************************************************************************
+ * Copyright (c) 2020 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0, or the MIT License which is
+ * available at https://opensource.org/licenses/MIT.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR MIT
+ ********************************************************************************/
+package org.eclipse.emfcloud.modelserver.emf.common;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import org.apache.log4j.Logger;
+import org.eclipse.emfcloud.modelserver.emf.configuration.ServerConfiguration;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+
+public class SchemaRepository {
```

Comment:
Should this class also be bound as a Singleton in the ModelServerModule? Currently it is only injected once and only in another singleton. Furthermore it does not hold any state besides the schema repository path. So I am opening this up for discussion, what do you think?

Reply:
That's a good point, thanks for noticing! I bound it as Singleton now. Also on creating other unit tests this came up as an issue and therefore makes a lot of sense.